client/src/main/com/sinch/sdk/domains/sms/models/v1/batches/internal/SMSCursorPageNavigator.java

```diff
@@ -0,0 +1,35 @@
+package com.sinch.sdk.domains.sms.models.v1.batches.internal;
+
+import com.sinch.sdk.core.models.pagination.PageNavigator;
+
+public class SMSCursorPageNavigator extends PageNavigator<Integer> {
+
+  private final Integer currentPage;
+  private final Integer pageSize;
+
+  public SMSCursorPageNavigator(Integer currentPage, Integer pageSize) {
+    super(null);
+    this.currentPage = currentPage;
+    this.pageSize = pageSize;
+  }
+
+  private Integer computeNextPageCursor() {
+    return null == pageSize || pageSize == 0 ? null : currentPage + 1;
+  }
+
+  @Override
+  public Integer getToken() {
```

Comment:
Will need to check it yes, but having to wait for a hard breaking change (V2)
The navigation used for auto-pagination is based onto this public class.
Changing it could have side effects onto any SDK user... even if chance are low to be used outside of internal SDK purpose